src/main/java/com/kwezal/bearinmind/core/user/service/UserService.java

```diff
@@ -49,28 +49,32 @@ public class UserService {
     private final UserValidationService userValidationService;
 
     @Transactional(readOnly = false)
-    public UserDto createUser(CreateUserDto createUserDto) {
+    public User createUser(final CreateUserDto createUserDto) {
         userValidationService.validateIfUserDoesNotExist(createUserDto.email(), ErrorCode.USER_EXISTS);
+
         final var applicationLocale = applicationConfig.getApplicationLocale();
         final var user = userMapper.map(createUserDto, applicationLocale);
         final var encodedPassword = passwordEncoder.encode(createUserDto.password());
+
         final var userCredentials = user.getUserCredentials();
         userCredentials.setPassword(encodedPassword);
         userCredentials.setRole(UserRole.STUDENT);
+
         userCredentialsRepository.save(userCredentials);
         userRepository.save(user);
-        return userMapper.mapToUserDto(user);
```

Comment:
Are we sure we want to avoid the mapping of user to DTO? I would say we shouldn't use an entity anywhere after it's inserted to the database.

Reply:
I don't see why we shouldn't use an entity for internal purposes. The reason was to avoid unnecessary mapping as we don't use this value in any HTTP response. Do you mean to prevent an entity in managed state from being updated by mistake in the future?

Reply:
We don't use it any HTTP response, but I think it has an impact of the redness. We use the mappers everywhere and I would say we should keep this as a pattern for us, because by doing this we keep a strong line between the business logic and the presentation layer. Of course this secures an unexpected situation when an entity can be changed by mistake as you mentioned.

Reply:
Alright, makes sense. Seems like it'll improve our code quality and the performance impact of unnecessary mapping is negligible. Will fix.